clients/src/main/java/org/apache/kafka/clients/consumer/internals/EventHandler.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.consumer.internals;
+
+/**
+ * The EventHandler interfaces the client and handler implementation.  The client should add an event for consumption,
+ * and try to poll for the responses.  Here, one could poll a response with type K, and add a request with type T.
+ * @param <T> Event request type
+ * @param <K> Event response type
+ */
+public interface EventHandler<T, K> {
+    public K poll();
+    public boolean add(T event);
```

Comment:
From our offline discussions, I think there are two channels of communicating "responses" back: 1) the caller send a request, along with a future, and then either sync or async waiting on the future, in this case it does not expect "polling" the response; 2) the caller send a request, and then either sync or async polling responses.

For commitSync, for example, since the caller only cares about a specific response, it may be not appropriate to try polling responses since there will be a lot of other responses getting polled but the caller does not care.

Could we add some javadocs clarifying which scenarios will leverage on each of the two?

Reply:
Absolutely.  I'm also thinking that the naming can be a bit misleading sometimes, because the ResponseEvents might not necessary be a "response".  As far as the implementation goes, I think they are mostly just errors and callback events.

Reply:
I was thinking something like RequestEvent -> ApplicationEvent and ResponseEvent -> NetworkEvent. Or something like that.

Reply:
Can we be more direct and call the ResponseEvent the BackgroundThreadEvent... because it really is coming from the background thread, and it's not necessary a NetworkEvent.  It can be just a trigger for the callback.

Reply:
That could work. Maybe `ApplicationEvent` and `BackgroundEvent`?

Reply:
I was thinking about `ConsumerEvent` or `CallerEvent`, and `CoordinatorEvent`, just to make the naming party worse :P